openshift/templates/nats.yaml

```diff
@@ -0,0 +1,208 @@
+---
+apiVersion: template.openshift.io/v1
+kind: Template
+metadata:
+  name: nats
+  annotations:
+    "openshift.io/display-name": nats
+parameters:
+  - name: APP_NAME
+    description: Application name (wps - wildfire predictive services)
+    value: wps
+  - name: SUFFIX
+    description: Deployment suffix, e.g. pr-###
+    required: true
+objects:
+  - apiVersion: v1
+    kind: ConfigMap
+    metadata:
+      name: nats-config
+      labels:
+        app: nats
+    data:
+      nats.conf: |-
+        pid_file: "/var/run/nats/nats.pid"
+        http: 8222
+        max_payload: 100Mb
+        max_pending: 120Mb
+        debug: false
+        trace: false
+        trace_verbose: false
+        server_name: $POD_NAME
+        cluster {
+            port: 6222
+            name: wps-nats-cluster
+            routes [
+               nats://nats-0.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+               nats://nats-1.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+               nats://nats-2.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+               nats://nats-3.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+               nats://nats-4.nats.${APP_NAME}-${SUFFIX}.svc.cluster.local:6222
+            ]
+            cluster_advertise: $CLUSTER_ADVERTISE
+            connect_retries: 50
+        }
+        leafnodes {
+          port: 7422
+        }
+        jetstream {
+          store_dir: /data
+          max_file: 5Gi
+        }
+  - apiVersion: v1
+    kind: Service
+    metadata:
+      name: nats
```

Comment:
You should be able to connect to this service, instead of the pods. At least within the cluster. So I think this is good to go. A pod running in the cluster should resolve "nats". This is how we connect to the patroni pods - via the service.

Reply:
Awesome, thanks!